bin/strata-client/src/args.rs

```diff
@@ -57,3 +62,114 @@ pub struct Args {
     #[argh(option, description = "database retry count")]
     pub db_retry_count: Option<u16>,
 }
+
+impl Args {
+    pub fn derive_config(&self) -> Result<Config, String> {
+        let args = self.clone();
+        Ok(Config {
+            bitcoind_rpc: BitcoindConfig {
+                rpc_url: args
+                    .bitcoind_host
+                    .ok_or_else(|| "args: no bitcoin --rpc-url provided".to_string())?,
+                rpc_user: args
+                    .bitcoind_user
+                    .ok_or_else(|| "args: no bitcoin --rpc-user provided".to_string())?,
+                rpc_password: args
+                    .bitcoind_password
+                    .ok_or_else(|| "args: no bitcoin --rpc-password provided".to_string())?,
+                network: args
+                    .network
+                    .ok_or_else(|| "args: no bitcoin --network provided".to_string())?,
+            },
+            client: ClientConfig {
+                rpc_host: args
+                    .rpc_host
+                    .ok_or_else(|| "args: no client --rpc-host provided".to_string())?,
+                rpc_port: args
+                    .rpc_port
+                    .ok_or_else(|| "args: no client --rpc-port provided".to_string())?,
+                datadir: args
+                    .datadir
+                    .ok_or_else(|| "args: no client --datadir provided".to_string())?,
+                client_mode: {
+                    if let Some(sequencer_key) = args.sequencer_key {
+                        ClientMode::Sequencer(SequencerConfig {
+                            sequencer_key,
+                            sequencer_bitcoin_address: args.sequencer_bitcoin_address,
+                        })
+                    } else if let Some(sequencer_rpc) = args.sequencer_rpc {
+                        ClientMode::FullNode(FullNodeConfig { sequencer_rpc })
+                    } else {
+                        return Err(
+                            "args: no client --sequencer-key or --sequencer-bitcion-address provided or --sequencer-rpc provided"
+                                .to_string(),
+                        );
+                    }
+                },
+                l2_blocks_fetch_limit: 1_000,
+                db_retry_count: 5,
+            },
+            sync: SyncConfig {
+                l1_follow_distance: 6,
+                max_reorg_depth: 4,
+                client_poll_dur_ms: 200,
+                client_checkpoint_interval: 10,
+            },
+            exec: ExecConfig {
+                reth: RethELConfig {
+                    rpc_url: args.reth_authrpc.unwrap_or("".to_string()), // TODO: sensible default
+                    secret: args.reth_jwtsecret.unwrap_or_default(),      /* TODO: probably
+                                                                           * secret should be
+                                                                           * Option */
```

Comment:
There is no sensible default here and we *always* need a secret.  We're not at the point where we can run without reth, so make these required.

Reply:
Oh these, I haven't had a change to go through in detail. These were just being moved around. Will make it required.